plugin/completion.py

```diff
@@ -1,36 +1,83 @@
 import sublime
 import sublime_plugin
-from .core.completion import parse_completion_response, format_completion
-from .core.configurations import is_supported_syntax
-from .core.documents import position_is_word
-from .core.edit import parse_text_edit
+
+from .core.protocol import Request, Range, InsertTextFormat
+from .core.settings import settings, client_configs
 from .core.logging import debug
-from .core.protocol import Request
+from .core.completion import parse_completion_response, format_completion
 from .core.registry import session_for_view, client_from_session, LSPViewEventListener
+from .core.configurations import is_supported_syntax
 from .core.sessions import Session
-from .core.settings import settings, client_configs
+from .core.edit import parse_text_edit
+from .core.views import range_to_region
 from .core.typing import Any, List, Dict, Tuple, Optional, Union
 from .core.views import text_document_position_params
 
 
-class CompletionState(object):
-    IDLE = 0
-    REQUESTING = 1
-    APPLYING = 2
-    CANCELLING = 3
+class LspSelectCompletionItemCommand(sublime_plugin.TextCommand):
+    def run(self, edit: Any, item: dict) -> None:
```

Comment:
You can rewrite this signature as `def run(self, edit: sublime.Edit, **item: Any) -> None:`,

and then in plugin/core/completion.py, you can pass the item dict as-is:

```
    return sublime.CompletionItem.command_completion(
        trigger,
        command="lsp_select_completion_item",
        args=item,
        annotation=annotation,
        kind=kind
    )
```
This saves us another key lookup, and it saves us a dict allocation.